### PR TITLE
Stdout logs improvements

### PR DESCRIPTION
**Pull request by tarekziade**
### Description

Introduces :

- JSON output by default (use `DEBUG=true` or the new `plain_logs` settings
- glob-style patterns for our log targets. e.g. you can now use `nucliadb_*`

### How was this PR tested?

Unit tests

### CODE REVIEW

File: nucliadb_node/src/telemetry.rs [hunk 1]
```diff
@@ -65,13 +66,46 @@ pub fn init_telemetry(settings: &Arc<Settings>) -> NodeResult<Option<ClientInitG
...
+                }
+                // starts with
+                if target.len() > 1 && target.ends_with('*') {
+                    target.truncate(target.len() - 1);
+                    if metadata_target.starts_with(&target) {
+                        println!("true");
+                        return true;
+                    }
+                }
+            }
```
Comment (hermeGarcia) [inline nucliadb_node/src/telemetry.rs:93]: What about moving some of this logic to `Settings`, that way we can do some previous computations that will render this filter lighter

**Comment by tarekziade** [comment]
@hermeGarcia one thing that frustrates me before we land the patch is that I was not able to extract the `filter` function builder out of the `stdout_layer` function so I can add some unit tests to verify the filtering code -- if you know how. Returning a `FilterFn` instance from an isolated function seems to be problematic in Rust because of how closures work.

File: nucliadb_node/src/telemetry.rs [hunk 2]
```diff
@@ -64,14 +68,100 @@ pub fn init_telemetry(settings: &Arc<Settings>) -> NodeResult<Option<ClientInitG
...
+
+        // match all
+        if self.all_target_level != LevelFilter::OFF && self.all_target_level >= *level {
+            return true;
+        }
 
+        let metadata_target = metadata.target();
+        // exact match
+        if let Some(log_level) = self.logs_map.get(metadata_target) {
+            if log_level >= level {
```
Comment (hermeGarcia) [inline nucliadb_node/src/telemetry.rs:111]:
```rust
if self.logs_map.get(metadata_target) >= Some(level) {
    return true;
}
```

File: nucliadb_node/src/telemetry.rs [hunk 3]
```diff
@@ -64,14 +68,100 @@ pub fn init_telemetry(settings: &Arc<Settings>) -> NodeResult<Option<ClientInitG
...
+                return true;
+            }
+        }
+        // prefixes match
+        for (prefix, log_level) in self.prefixes.iter() {
+            if log_level >= level && metadata_target.starts_with(prefix) {
+                return true;
+            }
+        }
+        false
```
Comment (hermeGarcia) [inline nucliadb_node/src/telemetry.rs:121]:
```rust 
        let possible_match = self
            .prefixes
            .iter()
            .find(|(prefix, log_level)| log_level >= level && metadata_target.starts_with(prefix));
        possible_match.is_some()
```

**Review by hermeGarcia — APPROVED** [review]
LGTM just a couple of comments

**Comment by tarekziade** [comment]
Example of output:

```
✗ RUST_LOG="nuclia*=DEBUG" ../target/release/node_reader 2>/dev/null

{"timestamp":"2023-10-02T10:03:38.472169Z","level":"INFO","fields":{"message":"telemetry to https://telemetry.nuclia.cloud/ is enabled."},"target":"nucliadb_telemetry::sender"}
```

**Pull request merged**